# 4. Написать функцию которая принимает на вход число от 1 до 100. Если число равно 13, функция поднимает исключительную
# ситуации ValueError иначе возвращает введенное число, возведенное в квадрат.
# Далее написать основной код программы. Пользователь вводит число. Введенное число передаем параметром в написанную
# функцию и печатаем результат, который вернула функция. Обработать возможность возникновения исключительной ситуации,
# которая поднимается внутри функции.
# Строго говоря, урок не дал мне полного понимания отличия ловли exeption от того же самого с помощью if
# Гугл тоже не советчик: If можно использовать, работает быстрее, но код менее читабелен. Catch
# рекомендовано использовать, но работает на порядок медленнее чем IF.


def except_func(in_number):
	if in_number == 13:
		raise ValueError('13')
	elif in_number <= 0 or in_number > 100:
		return f"Введено число {in_number}, нужно вводить число от 1 до 100"
	elif in_number in range(1, 13) or in_number in range(14, 101):
		return in_number ** 2
	else:
		return "Some other sh.t happens"


try:
	result = except_func(int(input('Введите число от 1 до 100 (кроме числа 13): ')))
except ValueError:
	print("Вводить следует только число от 1 до 100 (кроме числа 13)")
else:
	print(result)